oh_staff_ui/management/commands/import_projectitems.py

```diff
@@ -0,0 +1,101 @@
+        added_items = []
+        root_item = next(
+            item for item in projectitem_dicts if item["ARK"] == "21198/zz0008zh0f"
+        )
+        # create and save ProjectItem for root object
+        p = ProjectItem(
+            ark=root_item["ARK"],
+            create_date=self.format_date(root_item["CREATE_DATE"]),
+            created_by=self.get_or_create_user(root_item["CREATED_BY"]),
+            last_modified_date=self.format_date(root_item["LAST_MODIFIED_DATE"]),
+            last_modified_by=self.get_or_create_user(root_item["LAST_MODIFIED_BY"]),
+            sequence=self.format_sequence(root_item["SEQUENCE"]),
+            status=ItemStatus.objects.get(status=root_item["STATUS"]),
+            title=root_item["TITLE"],
+            type=ItemType.objects.get(type=root_item["TYPE"]),
+        )
+        p.save()
+        added_items.append(root_item)
+
+        # main logical loop
+        # until we've added all items, look in imported data for children of added items
+        # for each child we haven't seen already,
+        # save a PI object and add to list of completed items
+        while len(added_items) < len(projectitem_dicts):
+            for parent in added_items:
+                child_items = self.find_child_items(projectitem_dicts, parent["ARK"])
+                for child in child_items:
+                    if child not in added_items:
+                        self.add_projectitem(child)
+                        added_items.append(child)
+        total_added = len(added_items)
+        print(f"Finished adding {total_added} ProjectItems.")
+        total_projectitems = ProjectItem.objects.filter().count()
+        print(f"Number of ProjectItems in database: {total_projectitems}.")
+
+    def find_child_items(self, projectitem_dicts: list, parent_ark: str) -> list:
+        output_items = []
+        for line in projectitem_dicts:
+            if line["PARENT_ARK"] == parent_ark:
+                output_items.append(line)
+        return output_items
+
+    def add_projectitem(self, pi_dict: dict) -> list:
+        p = ProjectItem(
+            ark=pi_dict["ARK"],
+            create_date=self.format_date(pi_dict["CREATE_DATE"]),
+            created_by=self.get_or_create_user(pi_dict["CREATED_BY"]),
+            last_modified_date=self.format_date(pi_dict["LAST_MODIFIED_DATE"]),
+            last_modified_by=self.get_or_create_user(pi_dict["LAST_MODIFIED_BY"]),
+            parent=ProjectItem.objects.get(ark=pi_dict["PARENT_ARK"]),
+            sequence=self.format_sequence(pi_dict["SEQUENCE"]),
+            status=ItemStatus.objects.get(status=pi_dict["STATUS"]),
+            title=pi_dict["TITLE"],
+            type=ItemType.objects.get(type=pi_dict["TYPE"]),
+        )
+        p.save()
+
+    def format_date(self, date: str) -> str:
+        return datetime.datetime.strptime(date, "%d-%b-%y").strftime("%Y-%m-%d")
+
+    def format_sequence(self, seq) -> int:
```

Comment:
Is there a better approach here? The model setup requires an integer sequence number for every ProjectItem, but the exported data has many rows with blanks here. Should empty string map to 0 or 1, or should we update the model definition to make this field optional?

Reply:
Decision: Use 0 as value when blank; leave this as a required field.